### DISCUSSION TITLE
Is it possible to use this library to listen to a particular chart?

### DISCUSSION THREAD

**Discussion by lieyongchang** [gram-js/gramjs#251] [General]
as the title suggest, i am just wondering if it is possible to listen to only a chat if I have its chat id?

**Comment by painor** [comment]
yeah sure,
for example you do
```ts

client.addEventHandler(async (event) => {
     event.message.reply({message:"Chat id is "+event.message.chatId});
}, new NewMessage({chats: [-1001032212174]}));

```

**Reply by lieyongchang** [reply]
```
new NewMessage({chats: [-1001032212174]}));  // this is the other person chat id??

```

**Reply by rojvv** [reply]
It can be a person, a chat, a channel or whatever you like.

**Reply by lieyongchang** [reply]
Noted

**Reply by rojvv** [reply]
@lieyongchang please mark this as an answer.

**Reply by lieyongchang** [reply]
I cant cos i am not the owner of this repo

**Reply by rojvv** [reply]
Oh, I thought you can.

**Reply by lieyongchang** [reply]
Nope. I cant even close this discussion.

**Comment by Roebou1989** [comment]
@painor 
I have tried this, but I simply can't get this to work, not sure if I am missing anything.
For now, I would just like to console log the event currently, but it constantly does nothing.

````
await client.connect();

client.addEventHandler(async (event) => {
    console.log(event);
  }, new NewMessage({chats: [CHAT_USERNAME]}));

````

**Reply by painor** [reply]
you have to call a highlevel method so that telegram starts sending you events.
You can use client.getMe() for example after you do client.connect() (await both)
